### PR TITLE
Fix issue 284

### PR DESCRIPTION
**Pull request by mtoma**

### CODE REVIEW

File: docs/Architecture.md [hunk 1]
```diff
@@ -4,7 +4,7 @@ The Zonemaster *Backend* is a system for performing domain health checks and
 keeping records of performed domain health checks.
 
 A Zonemaster *Backend* system consists of at least three components: a
-single *Database*, a single *Worker* and one or more *Web backends*.
+single *Database*, a single *Test Agent* and one or more *Web backends*.
```
Comment (matsduf) [inline docs/Architecture.md]: I suggest that "Web backends" is replaced by "RPC API daemons".

File: docs/files-description.md [hunk 2]
```diff
@@ -16,30 +16,30 @@
...
+./lib/Zonemaster/Backend/TestAgent.pm
+    The TestAgent main module.
 
 ./script/execute_zonemaster_P10.pl
 ./script/execute_zonemaster_P5.pl
     The scripts to execute tests with differents priorities (application level priorities).
 
 ./script/execute_tests.pl
-    The main JobRunner entry point to execute from crontab.
+    The main TestAgent entry point to execute from crontab.
```
Comment (matsduf) [inline docs/files-description.md]: "TestAgent" --> "Test Agent"

**Comment by matsduf** [comment]
I just found two smaller things. Awaiting the review by @mattias-p too.

**Comment by mattias-p** [comment]
I found a bunch of files still mentioning Web Backend and Worker:
* README.md
* docs/API.md
* docs/GettingStarted.md
* docs/installation.md
* script/crontab_job_runner/execute_tests.pl
* share/zm-backend.sh
* share/zm-centos.sh

File: docs/API.md [hunk 3]
```diff
@@ -124,16 +124,16 @@ Properties:
 
 Basic data type: number
 
-This parameter is any integer that will be used by The Zonemaster Backend Worker daemon to sort the test requests from highest to lowest priority.
-This parameter will typically be used in a setup where a GUI will send requests to the Backend API and would like to get response as soon as possible while at the same time using the idle time for background batch testing.
+This parameter is any integer that will be used by The Zonemaster RPC API daemon daemon to sort the test requests from highest to lowest priority.
```
Comment (matsduf) [inline docs/API.md]: 1. "daemon daemon"
2. Is it really the RPC API daemon that sorts the requests? I thought it was the Test Agents.

File: docs/API.md [hunk 4]
```diff
@@ -124,16 +124,16 @@ Properties:
...
 
 ### Queue
 
 Basic data type: number
 
 This parameter allows an optional separation of testing in the same database. The default value for the queue is 0. It is closely related to the *lock_on_queue* parameter of the [ZONEMASTER] section of the backend_config.ini file.
-The typical use case for this parameter would be a setup with several separate Backends Worker daemons running on separate physical or virtual machines each one dedicated to a specific task, for example queue 0 for frontend tests and queue 1 dedicated to batch testing. Running several Backend Worker daemons on the same machine is currently not supported.
+The typical use case for this parameter would be a setup with several separate RPC API daemons running on separate physical or virtual machines each one dedicated to a specific task, for example queue 0 for frontend tests and queue 1 dedicated to batch testing. Running several RPC API daemons daemons on the same machine is currently not supported.
```
Comment (matsduf) [inline docs/API.md]: Shouldn't it be "several separate Test Agent daemons running on separate..."?

File: docs/installation.md [hunk 5]
```diff
@@ -72,10 +72,10 @@ service mysqld status
 > At this time there is no instruction for using SQLite on CentOS.
 >
 
-### 1.3 Installation of the backend
+### 1.3 Installation of the RPC API daemon
```
Comment (matsduf) [inline docs/installation.md]: I think it should be "Installation of Zonemaster Backend"

File: docs/installation.md [hunk 6]
```diff
@@ -142,7 +142,7 @@ user has just enough permissions to run the backend software.
 
 >
 > Note : Only run the above command during an initial installation of the
-> Zonemaster backend. If you do this on an existing system, you will wipe out the
+> Zonemaster RPC API daemon. If you do this on an existing system, you will wipe out the
```
Comment (matsduf) [inline docs/installation.md]: "Zonemaster RPC API daemon" --> "Zonemaster Backend"

File: docs/installation.md [hunk 7]
```diff
@@ -243,10 +243,10 @@ sudo apt-get install libdbd-pg-perl postgresql
 > At this time there is no instruction for using SQLite on Debian and Ubuntu.
 >
 
-### 2.3 Installation of the backend
+### 2.3 Installation of the RPC API daemon
```
Comment (matsduf) [inline docs/installation.md]: I think it should be "Installation of Zonemaster Backend"

File: share/zm-centos.sh [hunk 8]
```diff
@@ -8,9 +8,9 @@
 # Should-Stop:
 # Default-Start:     2 3 4 5
 # Default-Stop:      0 1 6
-# Short-Description: Start and stop the Zonemaster Web Backend demons
+# Short-Description: Start and stop the Zonemaster RPC API demons
```
Comment (matsduf) [inline share/zm-centos.sh]: This file will start both the RPC API daemon and the Test Agent daemon. Then the description should be
"Start and stop the Zonemaster Backend (RPC API daemon and Test Agent daemon)"

File: share/zm-backend.sh [hunk 9]
```diff
@@ -8,9 +8,9 @@
 # Should-Stop:
 # Default-Start:     2 3 4 5
 # Default-Stop:      0 1 6
-# Short-Description: Start and stop the Zonemaster Web Backend demons
+# Short-Description: Start and stop the Zonemaster RPC API demons
 # Description:       Control script for the two demon processes that
-#                    make up the Zonemaster Web Backend.
+#                    make up the Zonemaster RPC API.
```
Comment (matsduf) [inline share/zm-backend.sh]: "make up the Zonemaster Backend."

File: share/zm-centos.sh [hunk 10]
```diff
@@ -8,9 +8,9 @@
...
 # Description:       Control script for the two demon processes that
-#                    make up the Zonemaster Web Backend.
+#                    make up the Zonemaster RPC API.
```
Comment (matsduf) [inline share/zm-centos.sh]: "make up the Zonemaster Backend."

File: share/zonemaster_backend_testagent.conf [hunk 11]
```diff
@@ -1,11 +1,11 @@
-description "Zonemaster WebBackend Runner Daemon"
+description "Zonemaster Backend TestAgent Daemon"
```
Comment (matsduf) [inline share/zonemaster_backend_testagent.conf]: "TestAgent" --> "Test Agent"

File: MANIFEST.SKIP [hunk 12]
```diff
@@ -1,13 +1,13 @@
...
-Makefile$
-^blib
-^pm_to_blib
-^.*.bak
-^.*.old
-^t.*sessions
-^cover_db
-^.*\.log
-^.*\.swp$
+-^\.git\/
```
Comment (mattias-p) [inline MANIFEST.SKIP]: Please remove minus sign prefixes.

File: share/initial-mysql.sql [hunk 13]
```diff
@@ -19,7 +19,7 @@ CREATE TABLE test_results (
 	params blob NOT NULL,
 	results blob DEFAULT NULL,
     undelegated boolean NOT NULL DEFAULT false
-) Engine=InnoDB;
+) RPCAPI=InnoDB;
```
Comment (mattias-p) [inline share/initial-mysql.sql]: This looks wrong.

**Comment by matsduf** [comment]
@mtoma and @mattias-p, I suggest that we wait for Zonemaster::Engine to be published at CPAN before we merge this PR to see that travis is happy.

**Pull request merged**